git-repository/src/repository/state.rs

```diff
@@ -0,0 +1,42 @@
+use crate::RepositoryState;
+
+impl crate::Repository {
+    /// Returns the status of an in progress operation on a repository or [`RepositoryState::None`]
+    /// if nothing is happening.
+    pub fn in_progress_operation(&self) -> RepositoryState {
+        let repo_path = self.path();
```

Comment:
I'd call `repo_path` `git_dir`, which helps visualize that this is indeed happening within the `.git` directory and not within the working tree, which `repo_path` might also suggest.